scandium-core/src/main/java/org/eclipse/californium/scandium/util/ByteArrayUtils.java

```diff
@@ -143,23 +145,21 @@
 	/**
 	 * Takes a byte array and returns it HEX representation.
 	 * 
-	 * @param byteArray
-	 *            the byte array.
-	 * @return the HEX representation.
+	 * @param byteArray the byte array.
+	 * @return the HEX representation. Separated by spaces, e.g. "11 22 0A". if
+	 *         {@code null} or a empty array is provided, the result is "--".
 	 */
 	public static String toHexString(byte[] byteArray) {
 
 		if (byteArray != null && byteArray.length != 0) {
-
-			StringBuilder builder = new StringBuilder(byteArray.length * 3);
-			for (int i = 0; i < byteArray.length; i++) {
-				builder.append(String.format("%02X", 0xFF & byteArray[i]));
-
-				if (i < byteArray.length - 1) {
-					builder.append(' ');
-				}
+			char[] bytesHexadecimal = new char[byteArray.length * 3];
+			for (int src = 0, dest = 0; src < byteArray.length; src++) {
+				int value = byteArray[src] & 0xFF;
+				bytesHexadecimal[dest++] = BIN_TO_HEX_ARRAY[value >>> 4];
+				bytesHexadecimal[dest++] = BIN_TO_HEX_ARRAY[value & 0x0F];
+				bytesHexadecimal[dest++] = ' ';
```

Comment:
As far I as know this is only used for logging and for correlation context.
So we could remove the space and the "--" stuff ?